framework/classes/orm/attachment.php

```diff
@@ -46,7 +46,7 @@ public function __construct($from, $name, array $config)
         $this->cascade_delete  = array_key_exists('cascade_delete', $config)
             ? $config['cascade_delete'] : $this->cascade_delete;
 
-        $attachment_config = array_diff($config, array('key_from' => '', 'cascade_save' => '', 'cascade_delete' => ''));
+        $attachment_config = array_diff_assoc($config, array('key_from' => '', 'cascade_save' => '', 'cascade_delete' => ''));
```

Comment:
I think it should be neither *array_diff* nor *array_diff_key* but *array_diff_key* or even better *\Arr::filter_keys()* with third parameter to "true" : 
http://fuelphp.com/docs/classes/arr.html#/method_filter_keys